cli/build.rs

```diff
@@ -47,13 +50,13 @@ fn main() {
     }
 }
 
-fn get_git_hash() -> Option<String> {
-    if let Some(nix_hash) = std::env::var("NIX_JJ_GIT_HASH")
+fn get_git_has_from_nix() -> Option<String> {
+    std::env::var("NIX_JJ_GIT_HASH")
         .ok()
         .filter(|s| !s.is_empty())
-    {
-        return Some(nix_hash);
-    }
+}
+
+fn get_git_hash_from_jj() -> Option<String> {
     if let Ok(output) = Command::new("jj")
```

Comment:
nit: (at your option,) we can also turn this into let else.

Reply:
I rewrote this and the git version to a more functional style. Happy to drop those commits if that's too much!